### PR TITLE
ESQL: Make capabilities into an enum

### PR DESCRIPTION
**Pull request by alex-spies**
Should be merged after https://github.com/elastic/elasticsearch/pull/109926.

To avoid accidentally forgetting to add a capability, and to cut down on some boilerplate, turn `EsqlCapabilities` into an enum.

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java [hunk 1]
```diff
@@ -86,18 +77,10 @@ public class EsqlCapabilities {
...
         List<String> caps = new ArrayList<>();
-        caps.add(FN_CBRT);
-        caps.add(FN_IP_PREFIX);
-        caps.add(FN_SUBSTRING_EMPTY_NULL);
-        caps.add(AGG_TOP_LIST);
-        caps.add(ST_CENTROID_AGG_OPTIMIZED);
-        caps.add(METADATA_IGNORED_FIELD);
-        caps.add(FN_MV_APPEND);
-        caps.add(REPEAT);
-        caps.add(NAMED_POSITIONAL_PARAMETER);
```
Comment (luigidellaquila) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java:97]: I think this is missing in the enum

Comment (luigidellaquila) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java:97]: And any specific reason why `STRING_LITERAL_AUTO_CASTING_TO_DATETIME_ADD_SUB` is not in the enum either? I added it, but I see it's at the end of the list, so maybe I'm missing some context

**Review by luigidellaquila — COMMENTED** [review]
#109933 vs #109935, you win! (they are practically identical)

Looks good in general, just a couple of small comments

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java (hunk 1 above)
Comment (alex-spies) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlCapabilities.java:97]: No context, just missed it. Good catch!

**Comment by alex-spies** [comment]
> #109933 vs #109935, you win! (they are practically identical)
> 
> Looks good in general, just a couple of small comments

Haha, I win but yours is more complete (d'oh). Let's go with yours!

**Pull request closed without merging**